### ISSUE TITLE
Safari browser bug with safari.pushNotification.requestPermission not calling callback

### ISSUE THREAD

**Issue by jasonpang** [OneSignal/OneSignal-Website-SDK#126]
The Safari browser appears to be occasionally _missing_ the callback to `window.safari.pushNotification.requestPermission`. We depend on this callback to know that the user either granted or denied permissions. If the user granted permissions, we depend on this callback to kickoff the registration flow: registering with OneSignal, sending a welcome notification.

This bug seems to be most reproducible when clearing all data and testing:
#### Steps to Reproduce (high chance of reproduction, not guaranteed):
1. Open Safari.
2. Open the site you're testing on.
3. Visit Preferences -> Notifications, and Remove All notification permission entries.
4. Visit Preferences -> Privacy, and click Clear All Website Data...
5. On the site you're testing on, enable OneSignal logging with `OneSignal.log.setLevel('trace');`. This sets a value in LocalStorage to enable logging when you restart Safari later.
6. Restart Safari.
7. Visit your website. Observe how you're prompted to subscribe for notifications, but that clicking Allow does not kickstart the rest of the subscription process.

Affecting Safari 9.1 and possibly 10 (same issue on 10 when I first discovered it, but did not thoroughly test).

This issue does not happen as often after already getting one notification permission listed and when not clearing browser data.
#### Possible Solutions

Check on a timer after the call to `registerForPushNotifications()` that the user did grant permissions. Not an ideal solution, but there are very few resources to actually debug this issue (`/Library/Logs/apsd.log`) so we don't know the cause for it, and it's intermittent.

**Comment by jasonpang** [comment]
This user ran into the issue: https://app.intercom.io/a/apps/344a89aeac3f033e4dec4370781543b948aece6d/inbox/jason@onesignal.com/conversations/6297239402

**Comment by aslushnikov** [comment]
Hi,

Looks like I'm running into this issue on my website - https://stff.aslushnikov.com. Hitting the bell button does nothing in Safari.

Apparently, I'm not the only one - here's a fresh stackoverflow question about the same issue: https://stackoverflow.com/questions/40386947/onesignal-web-push-notification-on-safari-configuration-does-not-work#_=_

**Comment by jasonpang** [comment]
Hi there,

Sorry for the late reply. You're seeing the permission prompt right? If you refresh the page, does it work? 

If you're not seeing the permission prompt, then this bug isn't the issue you're experiencing. However, I am seeing the permission prompt for your site, so I'm guessing you're seeing it too.

**Comment by bluegalz02** [comment]
Hi, I have the same issue on Safari that when I click the subscription bell, it's not showing anything.

This is the log from OneSignal
<img width="1274" alt="screen shot 2016-11-17 at 16 25 32" src="https://cloud.githubusercontent.com/assets/20273907/20395268/9217bb96-ace2-11e6-9193-bfb33081b28a.png">

I'm using Safari 10.

**Comment by aslushnikov** [comment]
@jasonpang nah, I don't see any prompt unfortunately. My Safari version is Version 10.0.1 (11602.2.14.0.7). I asked a few friends to take a look, and it doesn't work for them in Safari either. 

Any suggestions?

**Comment by jasonpang** [comment]
Hey @aslushnikov,

Refreshing your Safari configuration, removing it and recreating it with the same settings seems to work for most users. Could you reach out to us on our chat support at onesignal.com, paste a thread to this and ask us to do this for you?

**Comment by jasonpang** [comment]
@bluegalz02,

Could you also reach out to us at onesignal.com, provide your OneSignal login email to us, and link to this thread and ask us to refresh your Safari config?

**Comment by jasonpang** [comment]
Someone else reported this. [Conversation](https://app.intercom.com/a/apps/344a89aeac3f033e4dec4370781543b948aece6d/inbox/web-push-team/conversations/7375603896)

**Comment by mesca** [comment]
I have the same problem: nothing happens when I click on the bell (no permission prompt).
I am using Safari version 10.0.3 (12602.4.8).
I don't think it comes from my side: I can't subscribe from the OneSignal homepage neither.

**Comment by volomike** [comment]
This may be a Safari bug after an OSX upgrade. I'm on High Sierra of OSX and encountered this problem. However, I resolved it by going to Safari menu > Preferences > Websites > Notifications, and unchecking that setting and then clicking another tab like Default to save that setting. Then, I cleared history. Then I went back to Safari menu > Preferences > Websites > Notifications and checked to get prompted on all notifications, and then clicked another tab like Default to save that setting. When I did that, the problem went away and Safari now prompted for notifications.

One more reason not to use Safari.
